private-templates-service/client/src/store/currentTemplate/actions.ts

```diff
@@ -69,15 +75,41 @@ function failureExistingTemplateUpdate(error: IncomingMessage): CurrentTemplateA
   };
 }
 
-export function updateTemplate(templateID: string, templateJSON: string, sampleDataJSON: string, templateName: string) {
+function requestTemplate(templateID: string): CurrentTemplateAction {
+  return {
+    type: GET_TEMPLATE,
+    text: "get single template",
+    templateID
+  }
+}
+
+function requestTemplateSuccess(template: Template, templateJSON: string, templateName: string, sampleDataJSON: string): CurrentTemplateAction {
+  return {
+    type: GET_TEMPLATE_SUCCESS,
+    text: "get single template success",
+    template,
+    templateJSON,
+    templateName,
+    sampleDataJSON,
+  }
+}
+
+function requestTemplateFailure(): CurrentTemplateAction {
+  return {
+    type: GET_TEMPLATE_FAILURE,
+    text: "get single template failure",
+  }
+}
+
+export function updateTemplate(templateID: string, templateJSON: string, sampleDataJSON: string, templateName: string, isPublished?: boolean) {
   return function (dispatch: any) {
     let api = new TemplateApi();
     api.setApiKey(0, "Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IkhsQzBSMTJza3hOWjFXUXdtak9GXzZ0X3RERSJ9.eyJhdWQiOiI4NjMxY2E0ZS1hMjVkLTQ5YWUtYmQ5OC1mZjNlMWRkZDQ0MTgiLCJpc3MiOiJodHRwczovL2xvZ2luLm1pY3Jvc29mdG9ubGluZS5jb20vNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3YyLjAiLCJpYXQiOjE1ODEzNjA2MTgsIm5iZiI6MTU4MTM2MDYxOCwiZXhwIjoxNTgxMzY0NTE4LCJhaW8iOiI0Mk5nWUdDVzk3Qm1NamRrY3pQK1pLSGZMeFlPQUE9PSIsImF6cCI6Ijg2MzFjYTRlLWEyNWQtNDlhZS1iZDk4LWZmM2UxZGRkNDQxOCIsImF6cGFjciI6IjEiLCJvaWQiOiI4YWRiZWRhNi00ZDdkLTQ1N2ItOTNkZC1jMTVmMWE1Y2NiMzEiLCJzdWIiOiI4YWRiZWRhNi00ZDdkLTQ1N2ItOTNkZC1jMTVmMWE1Y2NiMzEiLCJ0aWQiOiI3MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDciLCJ1dGkiOiJ6SDlkcHVQNnprcUJidC1MUnVGYkFBIiwidmVyIjoiMi4wIn0.WhdvkqeJymW-Ayeht6tafd8Z1muoMnyPhKoYq6KGrHdv6psfYQtmK0P0-TA5_zgOOHNNJvpVqH2LPnZTbpK4qgKLByR9umELHgD2FW9v5Djg1NAKqmQEGg_-Th__SGE3L9_WI58Wh0_Toh3f7fpLDzNBiC5iYDdGSaTilwxaYMGbXnJO2Y6Tow83GQATKGA3B27Xz2iBO9UFmEy9rte4DyLUAEIE6SCKwg-3YuD0zKfpgyd-lvhZZr37HeE9Y6hyOm_M4b_jwWI7oK0uZASuQer83W5jsSZNtiXg_T0euXJcvI9lL6R4oJDI_N6Y42RdDioIwX6FzOA4vRzTySZjBw");
 
     let newTemplate = new TemplateJSON();
     newTemplate.template = templateJSON;
     newTemplate.name = templateName;
-    newTemplate.isPublished = false;
+    newTemplate.isPublished = isPublished || false;
```

Comment:
can't this just be newTemplate.isPublished = isPublished ?

Reply:
newTemplate.isPublished is of type `boolean | undefined` so if its undefined it should be assigned false

Reply:
oh okay makes sense